lms/djangoapps/courseware/views/views.py

```diff
@@ -726,6 +726,12 @@ def _progress(request, course_key, student_id):
 
     # checking certificate generation configuration
     enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(student, course_key)
+
+    # If the learner is in verified modes and the student did not have
+    # their ID verified, we need to show message to ask learner to verify their ID first
+    must_be_id_verified = enrollment_mode in CourseMode.VERIFIED_MODES and \
```

Comment:
I like @adampalay's suggestion. Consider the following:
```py
can_request_certificate = SoftwareSecurePhotoVerification.user_is_verified(student) or enrollment_mode == CouseMode.HONOR
```
Then, in the template, flip the order of the conditional so that you show the button if `can_request_certificate` is True. Otherwise, don't.